contrib/nydusify/cmd/nydusify_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsPossibleValue(t *testing.T) {
+	value := "qwe"
+	list := []string{"abc", "qwe", "xyz"}
+	if !isPossibleValue(list, value) {
+		t.Errorf("isPossibleValue() expected return true, but got false")
+	}
+
+	// Failure situation
+	value2 := "vdf"
+	if isPossibleValue(list, value2) {
+		t.Errorf("isPossibleValue() expected return false, but got true")
+	}
+}
+
+func TestAddReferenceSuffix(t *testing.T) {
+	source := "localhost:5000/nginx:latest"
+	suffix := "-suffix"
+	target, err := addReferenceSuffix(source, suffix)
+	if err != nil {
+		t.Errorf("addReferenceSuffix(source, suffix) got err: %s", err)
+	}
+	if target != "localhost:5000/nginx:latest-suffix" {
+		t.Errorf("addReferenceSuffix(source, suffix) expected to return localhost:5000/nginx:latest-suffix, but got %s", target)
+	}
+
+	// Failure situation
+	source = "localhost:5000\nginx:latest"
+	suffix = "-suffix"
+	_, err = addReferenceSuffix(source, suffix)
+	if err == nil {
+		t.Errorf("addReferenceSuffix(source, suffix) expected to got error, but got nil")
+	} else if !strings.Contains(err.Error(), "invalid source image reference") {
+		t.Errorf("addReferenceSuffix(source, suffix) expected to got 'invalid source image reference', but got: %s", err)
+	}
+}
+
+func TestParseBackendConfig(t *testing.T) {
+	configJSON := `
+	{
+		"bucket_name": "test",
+		"endpoint": "region.oss.com",
+		"access_key_id": "testAK",
+		"access_key_secret": "testSK",
+		"meta_prefix": "meta",
+		"blob_prefix": "blob"
+	}`
+
+	if !json.Valid([]byte(configJSON)) {
+		t.Errorf("The input of parseBackendConfig() is not a valid json %s", configJSON)
+	}
+
+	file, err := ioutil.TempFile("", "nydusify-backend-config-test.json")
+	assert.Nil(t, err)
+	defer os.RemoveAll(file.Name())
+
+	_, err = file.WriteString(configJSON)
+	assert.Nil(t, err)
+	file.Sync()
+
+	resultJSON, err := parseBackendConfig("", file.Name())
+	assert.Nil(t, err)
+
+	if !json.Valid([]byte(resultJSON)) {
+		t.Errorf("The result of parseBackendConfig() is not a valid json %s", resultJSON)
+	}
+
+	if configJSON != resultJSON {
+		t.Errorf("The input and result of parseBackendConfig() are not equal %s \n\n %s", configJSON, resultJSON)
```

Comment:
In general, we don't need to provide detailed error info, otherwise the testing will become redundant.

Just use `require.Equal(t, configJSON, resultJSON)`.

Reply:
Fixed.